Zoom/Zoom7zUnarchiver.py

```diff
@@ -0,0 +1,175 @@
+        #pylint: disable=no-self-use
+        for format_str, extns in EXTNS.items():
+            for extn in extns:
+                if archive_path.endswith(extn):
+                    return format_str
+        # We found no known archive file extension if we got this far
+        return None
+
+    def main(self):
+        """Unarchive a file"""
+        # handle some defaults for archive_path and destination_path
+        archive_path = self.env.get("archive_path", self.env.get("pathname"))
+        if not archive_path:
+            raise ProcessorError(
+                "Expected an 'archive_path' input variable but none is set!")
+        destination_path = self.env.get(
+            "destination_path",
+            os.path.join(self.env["RECIPE_CACHE_DIR"], self.env["NAME"]))
+
+        # Create the directory if needed.
+        if not os.path.exists(destination_path):
+            try:
+                os.makedirs(destination_path)
+            except OSError as err:
+                raise ProcessorError("Can't create %s: %s"
+                                     % (destination_path, err.strerror))
+        elif self.env.get('purge_destination'):
+            for entry in os.listdir(destination_path):
+                path = os.path.join(destination_path, entry)
+                try:
+                    if os.path.isdir(path) and not os.path.islink(path):
+                        shutil.rmtree(path)
+                    else:
+                        os.unlink(path)
+                except OSError as err:
+                    raise ProcessorError("Can't remove %s: %s"
+                                         % (path, err.strerror))
+
+        fmt = self.env.get("archive_format")
+        if fmt is None:
+            fmt = self.get_archive_format(archive_path)
+            if not fmt:
+                raise ProcessorError(
+                    "Can't guess archive format for filename %s"
+                    % os.path.basename(archive_path))
+            self.output("Guessed archive format '%s' from filename %s"
+                        % (fmt, os.path.basename(archive_path)))
+        elif fmt not in EXTNS.keys():
+            raise ProcessorError(
+                "'%s' is not valid for the 'archive_format' variable. "
+                "Must be one of %s." % (fmt, ", ".join(EXTNS.keys())))
+
+        if fmt == "zip":
+            cmd = ["/usr/bin/ditto",
+                   "--noqtn",
+                   "-x",
+                   "-k",
+                   archive_path,
+                   destination_path]
+        elif fmt == "gzip":
+           cmd = ["/usr/bin/ditto",
+                  "--noqtn",
+                  "-x",
+                  archive_path,
+                  destination_path]
+        elif fmt == "7z":
+           cmd_7z = self.env.get("7z_unarchiver_path")
+           cmd = [cmd_7z,
+                  "x",
+                  archive_path,
+                  "-o"+destination_path]
```

Comment:
You've written this as though it could be a generic 7z unarchiver, but how can we be sure that all the 7z-unarchiving tools take the same command-line options?

I especially look sideways at the `"-o"+destination_path` construction, which is very non-standard. Does `"-o", destination_path` not work?